server/backend/api/app/routers/router_user_role_assignment.py

```diff
@@ -50,7 +46,9 @@ def create_user_role_assignment(
     db: Session = Depends(database.get_db),
     token_claims: dict = Depends(jwt_validation.validate_token),
     requester: Requester = Depends(get_current_requester),
-    target_user: TargetUser = Depends(target_user_bceid_search)
+    target_user: TargetUser = Depends(
+        enforce_bceid_by_same_org_guard
```

Comment:
so I let the enforce_bceid_by_same_org_guard to return the target user, and move it from router dependency to here, so we don't need to call the idim service twice

Reply:
It's the magic from "Depends", if the same depends being referenced more than once, it will not evaluate again.
So you don't need to move that validation to the parameters of the function. 
It simply make sense to be on "@router.post(dependencies=)" before any function execution. If moving to the function parameters as one of the dependency, it probably does not matter in this case, but that's already go beyond the life cycle of route and step into the route handler (which is the function). 

At the parameter you can, as it was, depends on "get_verified_target_user" and it will not call again externally (since it is a Depends).

Reply:
oh yes Ian, I remember you explained to me before, sorry I forgot, I'll change it back, thanks!!